sdk/openai/Azure.AI.OpenAI.Assistants/src/Custom/CreateFileRequest.Serialization.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System.Collections.Generic;
+using Azure.Core;
+
+namespace Azure.AI.OpenAI.Assistants;
+
+internal partial class CreateFileRequest : IUtf8JsonSerializable
+{
+    /*
+     * CUSTOM CODE DESCRIPTION:
+     *
+     * Manual, custom multipart/form-data serialization needed.
+     *
+     */
+
+    internal virtual RequestContent ToRequestContent()
+    {
+        MultipartFormDataContent content = new();
+
+        content.Add(MultipartContent.Create(Purpose.ToString()), "purpose", new Dictionary<string, string>());
+        content.Add(MultipartContent.Create(Data), new Dictionary<string, string>()
+        {
+            ["Content-Disposition"] = $"form-data; name=file; filename={(string.IsNullOrEmpty(Filename) ? "file" : Filename)}",
```

Comment:
nit: If I understand correctly, you don't have to specify the "Content-Disposition" header; it gets added automatically. You can specify the "name" and "filename" using the following overload of the `Add` method of the `MultipartFormDataContent` class:
`Add(RequestContent content, string name, string fileName, Dictionary<string, string> headers)`